src/Fabulous.Avalonia/AppBuilderExtensions.fs

```diff
@@ -0,0 +1,97 @@
+namespace Fabulous.Avalonia
+
+open Avalonia
+open Avalonia.Styling
+open Fabulous
+
+#if IOS
+type SingleViewLifetime() =
+    member val View: Avalonia.iOS.AvaloniaView = null with get, set
+
+    interface Avalonia.Controls.ApplicationLifetimes.ISingleViewApplicationLifetime with
+        member this.MainView
+            with get () = this.View.Content
+            and set value =
+                if this.View <> null then
+                    this.View.Content <- value
+#endif
+
+[<AutoOpen>]
+module AppBuilderExtensions =
+    type AppBuilder with
+
+        static member inline private UseFabulousApp(canReuseView, logger, [<InlineIfLambda>] viewFn: unit -> Widget, theme: IStyle) =
+            let builder =
+                AppBuilder.Configure(fun () ->
+                    let app =
+                        FabApplication(
+                            OnFrameworkInitialized =
+                                fun app ->
+                                    let widget = viewFn()
+
+                                    let treeContext: ViewTreeContext =
+                                        { CanReuseView = canReuseView
+                                          Logger = logger
+                                          Dispatch = ignore
+                                          GetViewNode = ViewNode.get
+                                          GetComponent = Component.get }
+
+                                    let def = WidgetDefinitionStore.get widget.Key
+                                    def.AttachView(widget, treeContext, ValueNone, app) |> ignore
+                        )
+
+                    app.Styles.Add(theme)
+                    app)
+
+#if !ANDROID && !IOS
+            builder.UseAvaloniaNative().UseSkia()
```

Comment:
This is a test. Should be `.UsePlatformDetect()`

Reply:
Seems like Avalonia is not happy with the way we are initialising, because I get different runtime errors is we use `UseSkia()` vs `.UsePlatformDetect()`

Reply:
Also found a recent PR https://github.com/AvaloniaUI/Avalonia/pull/14267 merged that seems remove the need for AttachView 🧐

Reply:
Something seems to be calling `Dispatcher.UIThread` before it is initialized by `.UsePlatformDetect()`

Reply:
but Fabulous.Avalonia is not doing anything like that

Reply:
We could try without the UseFabulous type extension 🤔